libfoundation/src/foundation-private.h

```diff
@@ -624,6 +624,9 @@ __MCAssertResolvedTypeInfo(MCTypeInfoRef x, bool (*p)(MCTypeInfoRef))
 
 #define __MCAssertValueType(x,T) MCAssert(MCValueGetTypeCode(x) == kMCValueTypeCode##T)
 
+// A valid ValueRef must have references > 0 and flags != -1
+#define __MCAssertIsValue(x)    MCAssert(((__MCValue *)x) -> flags != 0xffffffff && ((__MCValue *)x) -> references > 0);
```

Comment:
I think it might be worth making `__MCAssertIsValue(x)` also assert that `x != nil` because `nil` is not a valid valueref.